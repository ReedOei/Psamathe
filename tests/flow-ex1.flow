contract Main {}

;

var A : ! nat
var B : ! nat
var C : ! nat
var D : any set ! string
var E : any set ! string

2 --> A
A --[ 1 ]-> B
B --> C

A --[ 1 ]-> C --> B

"foo" --> D
"bar" --> D
D --[ "foo" ]-> E

A --> consume
B --[ 1 ]-> consume

