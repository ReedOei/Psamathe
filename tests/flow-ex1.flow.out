<generatedTop>
  <common>
    <k>
      .
    </k>
    <contracts>
      Main |-> contract Main { type LunchName is nonfungible  consumable  asset  .Decls string  .Decls }
    </contracts>
    <types>
      Main . LunchName |-> modified ( SetItem ( asset )
      SetItem ( consumable )
      SetItem ( nonfungible ) , string )
      address |-> modified ( SetItem ( consumable )
      SetItem ( nonfungible ) , address )
      bool |-> modified ( SetItem ( consumable )
      SetItem ( nonfungible ) , bool )
      nat |-> modified ( SetItem ( consumable )
      SetItem ( fungible ) , nat )
      string |-> modified ( SetItem ( consumable )
      SetItem ( nonfungible ) , string )
    </types>
    <transactions>
      .Map
    </transactions>
    <log>
      ListItem ( #ostream ( 1 ) )
      ListItem ( "on" )
      ListItem ( #buffer ( "" ) )
    </log>
  </common>
  <storage>
    A |-> resource ( one nat , 0 )
    B |-> resource ( one nat , 1 )
    C |-> resource ( one nat , 0 )
    D |-> resource ( set string , SetItem ( "bar" ) )
    E |-> resource ( set string , SetItem ( "foo" ) )
  </storage>
  <localTypeEnv>
    A |-> one nat
    B |-> one nat
    C |-> one nat
    D |-> set string
    E |-> set string
    x |-> one Main
  </localTypeEnv>
  <currentContract>
    NoContract
  </currentContract>
  <activations>
    .Set
  </activations>
</generatedTop>

